src/engine/blocks.js

```diff
@@ -511,7 +523,11 @@ class Blocks {
     _getBlockParams (block) {
         const params = {};
         for (const key in block.fields) {
-            params[key] = block.fields[key].value;
```

Comment:
Noticed a problem here when connecting this with the GUI. The variable ID is being reported in the monitor instead of the variable name. This actually should use `.value` because it is used by the monitor reporting so it should include the name of the variable, not the ID. Sorry there is no useful indication of that.